src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -92,6 +92,7 @@
   private static final ExecutorService METRICS_GRABBER_EXECUTOR = Executors.newFixedThreadPool(10);
   private static final long METRICS_POLL_INTERVAL_MS = TimeUnit.SECONDS.toMillis(5);
   private static final long METRICS_MAX_WAIT_MS = TimeUnit.MINUTES.toMillis(2);
+  private static final UUID GLOBAL_LOCK_ID = UUID.fromString("35446a30-86ed-11e9-bc42-526af7764f64");
```

Comment:
Given that `SegmentRunner. processTriggeredSegment(..)` is called when the lock is held, does this not have the consequence that only segment will ever run in Reaper at any one time, regardless of how many clusters are registered in Reaper or how many Reaper instances there are?

Reply:
Good point, it'll actually be only one segment that can *start* at once, not run.
Still it's suboptimal, I'll adjust this so we have one lock per cluster.

Reply:
I've changed it to a cluster wide lock

Reply:
> Good point, it'll actually be only one segment that can start at once, not run.

Are you sure about that?
Because it looks like `SegmentRunner. processTriggeredSegment(..)`  waits until the segment is complete.

Reply:
Yes, because one SegmentRunner instance gets created per segment.

Reply:
> Yes, because one SegmentRunner instance gets created per segment.

 Yes, but the global/cluster lock is not released until after `SegmentRunner. processTriggeredSegment(..)`. The instance of the SegmentRunner is irrelevant.

Reply:
I disagree, we just need to release the lock earlier (before getting into `processTriggeredSegment()`).

Reply:
I'm  `-1` on expanding the lock to be global or cluster because it breaks the premise of introducing the ~double-locking-idiom.

The purpose of the `WARN` log on https://github.com/thelastpickle/cassandra-reaper/pull/622/files#diff-e6aa6503be815f976c3ed7a845621f4aR297 was to prove or disprove the need for the LWT locking.
But by making the locking global or cluster-based the locking is going to be overly eager and won't prove or disprove anything. 

Can the lock be returned to how it was, so to guarantee the transition from `RepairSegment.State.STARTED` to `RepairSegment.State.RUNNING`.
The second aspect to this code (ie the result of the `canRepair(..)` method) is the load of the nodes that the token range has replicas for, and those jmx calls are not bound by the LWT transaction anyway, regardless of how broad the lock is made.

Reply:
> Can the lock be returned to how it was, so to guarantee the transition from RepairSegment.State.STARTED to RepairSegment.State.RUNNING.

Previously the lock was on the whole repair. That was the best we could do because we need to lock against all the nodes involved in the token range at hand. There's no way to achieve that without executing three separate LWTs locks. 

I'm `-0` on a cluster-wide lock, and would rather it returned to a `repairId` lock as it was. 
Otherwise, either way, the first `WARN` log it not going to be specific enough to tell us the lock is never needed, as either repair or cluster based it is too broad. The second `WARN` log on line 310 should tell us though.